api.py

```diff
@@ -506,18 +506,73 @@ def ingest_samples(dataset_id, sample_list, **kwargs):
     return _ingest_entity_set(dataset_id, entity_set, **kwargs)
 
 
-def ingest_entities(dataset_id, data, **kwargs):
+def convert_entities_to_entity_set(entity_list):
+    conduce_keys = ['id', 'kind', 'point', 'path', 'polygon']
+    entities = []
+    ids = set()
+    for idx, ent in enumerate(entity_list):
+        if not 'id' in ent:
+            raise ValueError('Error processing entity at index {}. Entities must include an ID.'.format(idx), ent)
+        if not 'kind' in ent:
+            raise ValueError('Error processing entity at index {}. Entities must include a kind field.'.format(idx), ent)
+
+        if ent['id'] is None or len(str(ent['id'])) == 0:
+            raise ValueError('Error processing entity at index {}. Invalid ID.'.format(idx), ent)
+        if ent['id'] in ids:
+            raise ValueError('Error processing entity at index{}. Non-Unique ID'.format(idx), ent)
+        if ent['kind'] is None or len(ent['kind']) == 0:
+            raise ValueError('Error processing entity at index {}. Invalid kind.'.format(idx), ent)
+        if ent.get('time') is not None:
+            raise KeyError('Error processing entity at index {}. Timeless entities should not set time.'.format(idx), ent)
+
+        coordinates = convert_geometries(ent)
+        if coordinates == []:
+            raise ValueError('Error processing entity at index {}.  Entities must define a location (point, path, or polygon)'.format(idx), ent)
+
+        ids.add(ent['id'])
+        ent['_kind'] = ent['kind']
+        attribute_keys = [key for key in ent.keys() if key not in conduce_keys]
+        attributes = util.get_attributes(attribute_keys, ent)
+
+        entities.append({
+            'identity': str(ent['id']),
+            'kind': ent['kind'],
+            'timestamp_ms': util.get_default('timestamp_ms'),
+            'endtime_ms': util.get_default('endtime_ms'),
+            'path': coordinates,
+            'attrs': attributes
+        })
+
+    return {'entities': entities}
+
+
+def ingest_entities(dataset_id, entity_list, **kwargs):
     """
     Upload :ref:`entities <conduce-entities>` to the Conduce datastore.
 
     A convenience method that adds entities to the Conduce datastore and waits for the job to complete. This function POSTs an entity set to Conduce and :py:func:`wait_for_job` until the ingest job completes.
 
+    Each element of ``entity_list`` must have the following structure::
+
+        {
+            "id": <string>,
+            "kind": <string>,
+            "point": {
```

Comment:
Are we only allowing points at this time?

Reply:
No, we also allow paths and polygons.  However, I don't want to document those alternate structures here.  Would you be satisfied if I changed the language from "must" to something that indicates this is an example of a valid structure.

I link to the conduce-entities document above so I don't need to do it again here, I think.

Reply:
I've made changes to address your comment.